addons/mrp/models/mrp_production.py

```diff
@@ -674,10 +678,66 @@ def _get_move_raw_values(self, product_id, product_uom_qty, product_uom, operati
+            produced = sum(move_finished_ids.mapped('quantity_done'))
+            if produced > new_qty:
+                precision_digits = - int(log10(move_finished_ids.product_uom.rounding))
+                produced = float_repr(produced, precision_digits=precision_digits)
+                new_qty = float_repr(new_qty, precision_digits=precision_digits)
+                raise UserError(_("You have already processed %s. Please input a quantity higher than %s ") % (produced, new_qty))
+
+            previous_qty = production.product_qty
+            production.product_qty = new_qty
+
+            posted_move_finished_ids = move_finished_ids.filtered(lambda m: m.state == 'done')
+            posted_qty = sum(posted_move_finished_ids.mapped('product_uom_qty'))
+            ongoing_qty = production.product_qty - posted_qty
+
+            ongoing_qty_bom_uom = production.product_uom_id._compute_quantity(ongoing_qty, production.bom_id.product_uom_id)
+            factor = ongoing_qty_bom_uom / production.bom_id.product_qty
+            boms, lines = production.bom_id.explode(production.product_id, factor, picking_type=production.bom_id.picking_type_id)
+
+            documents = defaultdict(list)
+            for line, line_values in lines:
+                if line.child_bom_id and line.child_bom_id.type == 'phantom' or\
+                        line.product_id.type not in ['product', 'consu']:
+                    continue
+                move = production.move_raw_ids.filtered(lambda x: x.bom_line_id.id == line.id and x.state not in ('done', 'cancel'))
+                if move:
+                    move = move[0]
+                    old_qty = move.product_uom_qty
+                    iterate_key = move._get_document_iterate_key()
+                    if iterate_key:
+                        document = self.env['stock.picking']._log_activity_get_documents({move: (line_values['qty'], old_qty)}, iterate_key, 'UP')
+                        for key, value in document.items():
+                            documents[key] += [value]
+                production._update_raw_move(line, line_values)
+            if documents:
+                production._log_manufacture_exception(documents)
+            ongoing_move_raw_ids = production.move_raw_ids.filtered(lambda m: m.state not in ('done', 'cancel'))
+
+            finished_moves_modification = production._update_finished_moves(ongoing_qty, previous_qty)
+            production._log_downside_manufactured_quantity(finished_moves_modification)
+            ongoing_move_raw_ids._action_assign()
+
+            qty_per_operation = {}
+            for bom, bom_data in boms:
+                for operation in bom.routing_id.operation_ids:
+                    qty_per_operation[operation.id] = bom_data['qty']
+            if production.state == 'planned':
+                production.button_unplan()
+                production.button_plan()
```

Comment:
tendu non?

Reply:
Sinon le planning n'est pas update 

OMA - 25th February 2020
1. Update button on the MO: updates correctly the expected duration on the work orders but doesn't replan